x/token/keeper/msg_server_test.go

```diff
@@ -0,0 +1,125 @@
+package keeper_test
+
+import (
+	sdk "github.com/line/lbm-sdk/types"
+	"github.com/line/lbm-sdk/x/token"
+)
+
+func (s *KeeperTestSuite) TestMsgTransfer() {
+	testCases := map[string]struct{
+		classId string
+		amount sdk.Int
+		valid bool
+	}{
+		"valid request": {
+			classId: s.classID,
+			amount: s.balance,
+			valid: true,
+		},
+		"insufficient funds (amount)": {
+			classId: s.classID,
+			amount: s.balance.Add(sdk.OneInt()),
+			valid: false,
+		},
+		"insufficient funds (no such a class)": {
+			classId: "typo",
+			amount: s.balance,
+			valid: false,
+		},
+	}
+
+	for name, tc := range testCases {
+		s.Run(name, func() {
+			req := &token.MsgTransfer{
+				ClassId: tc.classId,
+				From: s.vendor.String(),
+				To: s.customer.String(),
+				Amount: tc.amount,
+			}
+			res, err := s.msgServer.Transfer(s.goCtx, req)
+			if !tc.valid {
+				s.Require().Error(err)
+				return
+			}
+			s.Require().NotNil(res)
+		})
+	}
+}
+
+func (s *KeeperTestSuite) TestMsgTransferFrom() {
+	testCases := map[string]struct{
+		classId string
+		proxy sdk.AccAddress
+		from sdk.AccAddress
+		valid bool
+	}{
+		"valid request": {
+			classId: s.classID,
+			proxy: s.operator,
+			from: s.customer,
+			valid: true,
+		},
+		"not approved": {
+			classId: s.classID,
+			proxy: s.vendor,
+			from: s.customer,
+			valid: false,
+		},
+	}
+
+	for name, tc := range testCases {
+		s.Run(name, func() {
+			req := &token.MsgTransferFrom{
+				ClassId: tc.classId,
+				Proxy: tc.proxy.String(),
+				From: tc.from.String(),
+				To: s.vendor.String(),
+				Amount: s.balance,
+			}
+			res, err := s.msgServer.TransferFrom(s.goCtx, req)
+			if !tc.valid {
+				s.Require().Error(err)
+				return
+			}
+			s.Require().NotNil(res)
+		})
+	}
+}
+
+func (s *KeeperTestSuite) TestMsgApprove() {
+	testCases := map[string]struct{
+		classId string
+		approver sdk.AccAddress
+		proxy sdk.AccAddress
+		valid bool
+	}{
+		"valid request": {
+			classId: s.classID,
+			approver: s.customer,
+			proxy: s.vendor,
+			valid: true,
+		},
+		"already approved": {
+			classId: s.classID,
+			approver: s.customer,
+			proxy: s.operator,
+			valid: false,
+		},
+	}
+
+	for name, tc := range testCases {
+		s.Run(name, func() {
+			req := &token.MsgApprove{
+				ClassId: tc.classId,
+				Approver: tc.approver.String(),
+				Proxy: tc.proxy.String(),
+			}
+			res, err := s.msgServer.Approve(s.goCtx, req)
+			if !tc.valid {
+				s.Require().Error(err)
+				return
+			}
+			s.Require().NotNil(res)
+		})
+	}
+}
```

Comment:
Why  doesn't the unittest of `Issue` and `Grant`, etc exist?

Reply:
I've added it!